lib/charms/loki_k8s/v0/loki_push_api.py

```diff
@@ -481,24 +482,29 @@ def _alert_rules_error(self, event):
 DEFAULT_ALERT_RULES_RELATIVE_PATH = "./src/loki_alert_rules"
 DEFAULT_LOG_PROXY_RELATION_NAME = "log-proxy"
 
-PROMTAIL_BINARY_ZIP_URL = (
-    "https://github.com/grafana/loki/releases/download/v2.4.1/promtail-linux-amd64.zip"
-)
-
+PROMTAIL_BASE_URL = "https://github.com/grafana/loki/releases/download"
+# To update Promtail version you only need to change the PROMTAIL_VERSION and
+# update all sha256 sums in PROMTAIL_BINARIES. To support a new architecture
+# you only need to add a new key value pair for the architecture in PROMTAIL_BINARIES.
+PROMTAIL_VERSION = "v2.4.1"
+PROMTAIL_BINARIES = {
+    "amd64": {
+        "filename": "promtail-linux-amd64",
+        "zipsha": "978391a174e71cfef444ab9dc012f95d5d7eae0d682eaf1da2ea18f793452031",
+        "binsha": "00ed6a4b899698abc97d471c483a6a7e7c95e761714f872eb8d6ffd45f3d32e6",
+    },
+    "arm64": {
+        "filename": "prometail-linux-arm64",
+        "zipsha": "89f66a5b0daf47a1d7cb4d7db1a82278976384c7f60138b68abb12c72e332ce2",
+        "binsha": "b1590454a6bf5a6d6188ad54060cd377b447e6dbfda6d1376561ab066731634d",
+    },
+}
```

Comment:
@balbirthomas I think we should close [this PR](https://github.com/canonical/loki-k8s-operator/pull/148) and implement changes made there here.

For instance, using our own statically compiled Promtail binaries: https://github.com/canonical/loki-k8s-operator/releases/tag/promtail-v2.5.0

Reply:
Ok. Will do it. I think it should be straight forward - just changing the base URL, filenames and shasums. Will test and then push tomorrow.